lighthouse-core/gather/driver.js

```diff
@@ -852,9 +852,20 @@ class Driver {
     return collectUsage;
   }
 
-  blockUrlPatterns(urlPatterns) {
-    const promiseArr = urlPatterns.map(url => this.sendCommand('Network.addBlockedURL', {url}));
-    return Promise.all(promiseArr);
+  /**
+   * This method sets the URL patterns to be blocked and is called at the beginning of each pass.
+   * No "clearing" is done at the end of the pass since the next pass will set to [] if necessary.
+   * @param {!Array<string>} urls
```

Comment:
maybe worth adding `param` doc similar to what the protocol docs has? ("URL patterns to block. Wildcards ('*') are allowed.")